### DISCUSSION TITLE
init: (164) ERROR: LogException:33: TELEMETRY: (null) No such file or directory @d:\os\src\onecore\VM\linux\mountutil\mountutilcpp.h:19 (MountEnum)

### DISCUSSION THREAD

**Discussion by cerebrate** [microsoft/WSL#7323] [Q&A]
So, I keep seeing this error

```
init: (164) ERROR: LogException:33: TELEMETRY: (null) No such file or directory @d:\os\src\onecore\VM\linux\mountutil\mountutilcpp.h:19 (MountEnum)
```

showing up in my kernel messages/system log periodically, along with a slightly rarer cousin:

```
init: (164) ERROR: LogException:33: TELEMETRY: (null) Invalid argument @d:\os\src\onecore\VM\linux\mountutil\mountutilcpp.h:19 (MountEnum)
```

I don't want to open an issue for this because I'm only getting it on my main machine with custom kernel, etc., and haven't been able to repro it yet on my clean-and-standard kernel system - but while I'm working on that repro case, I thought I'd throw this one out here. Anyone else seeing this (apart from #6895, that is)? Better yet, anyone have a non-obvious thought about possible causes?

(I've got an ETL file, but I've been through that myself and didn't spot anything particularly relevant. To my naive eye, anyway.)

**Comment by benhillis** [comment]
Yep this typically happens with short-lived processes. Nothing you need to be worried about, I need to clean this up... Thanks for reporting.

**Reply by cerebrate** [reply]
Ah, good. Thanks!

Just to mention a related weird thing I noticed today: while these messages are logged at _err_ priority, there seems to be a blank line logged after each of them which - in logging daemons rather than _dmesg_ - comes out as a second message, source unknown, whose priority is all the way up at _alert_ .

A quite alarming discovery when your new configuration sends those straight to _wall_ by default!

**Reply by benhillis** [reply]
Interesting, we do print a newline at the end of the message, maybe we shouldn't be...

**Reply by cerebrate** [reply]
I did a little experiment with this: writing to /dev/kmsg, it looks like if I embed the newline in the message in one write, it does get treated as part of the message. Both:

```
echo "<3>This is a test (no newline)." > /dev/kmsg
echo "<3>This is a test (newline).\n" > /dev/kmsg
```

Come through to systemd-journald as a single entry of (as specified) priority 3/_err_ , with (in the latter case) a newline embedded:

```
Aug 19 12:36:00 pallas-wsl unknown: This is a test (no newline).
Aug 19 12:37:13 pallas-wsl unknown: This is a test (newline).

Aug 19 12:37:19 pallas-wsl unknown: This is a test (newline).

```

Maybe that newline/blank line is being treated as a separate write to /dev/kmsg? (If I'm reading the kernel source right, the config file for the WSL kernel does assign messages without the magic number in front 1/_alert_ priority rather than the default 4/_warn_, which explains that part of it.)
